src/Vault.ts

```diff
@@ -60,20 +60,25 @@ export class Vault {
         return this.config.vaultToken;
     }
 
-    public async read(path: string | string[], parameters?: HTTPGETParameters, acceptedReturnCodes?: number[]): Promise<any> {
-        return this.request("GET", path, {}, parameters, acceptedReturnCodes);
+    public async read(
+        path: string | string[],
+        parameters?: HTTPGETParameters,
+        retryWithTokenRenew?: boolean,
+        acceptedReturnCodes?: number[],
```

Comment:
Since we're breaking the interface of this library anyway, maybe just add an `options` parameter of type `{retryWithTokenRenew?: boolean, acceptedReturnCodes?: number[]}` which can be extended dynamically?